concepts/while-loops/about.md

```diff
@@ -0,0 +1,147 @@
+This is not a good practice and should be avoided.
+
+## While Loops
+
+A while loop is a control structure that allows code to be executed repeatedly based on a given condition.
+The code within a while loop will continue to execute until the condition evaluates to falsey.
+While loops are often used when the number of iterations is not known beforehand but can be used in any situation where a loop is needed.
+
+The condition is evaluated before the code within the loop is executed, which means that if the condition is true the code within the loop will never be executed.
+
+```crystal
+i = 0
+while i < 3
+  puts i
+  i += 1
+end
+
+# Output:
+# 0
+# 1
+# 2
+```
+
+When wanting to iterate over a string, you can loop over the length of the string.
+
+```crystal
+str = "Hello"
+i = 0
+while i < str.size
+  puts str[i]
+  i += 1
+end
+
+# Output:
+# H
+# e
+# l
+# l
+# o
+```
+
+## Until Loops
+
+Until loops work the same as while loops, except that the code within the loop will continue to execute until the condition evaluates to truthy.
+It is the same as the difference between `if` and `unless`.
+
+```crystal
+i = 0
+until i == 3
+  puts i
+  i += 1
+end
+```
+
+## Break
+
+The `break` keyword can be used to exit a loop early.
+This can be convenient when you want to exit a loop early under certain conditions.
+
+```crystal
+i = 0
+while i < 3
+  puts i
+  i += 1
+  if i == 2
+    break
+  emd
+end
+
+# Output:
+# 0
+# 1
+```
+
+## Next
+
+The `next` keyword can be used to skip to the next iteration of a loop.
+This can be convenient when you want to skip certain iterations of a loop.
+
+```crystal
+i = 0
+while i < 3
+  i += 1
+  if i == 2
+    next
+  end
+  puts i
+end
+
+# Output:
+# 1
+# 3
+```
+
+## Types under looping
+
+The type of the variable can be different under looping.
+In the following example, the type of `a` is `Int32 | String` because it can be `Int32` or `String` under the first part of the loop.
+The variable `a` will only carry the type `Int32 | String` under the first part of the loop, since when the loop starts it holds `Int32` and under the 2nd iteration it holds `String`.
+Exactly what this multiple type system is, will be explained in later concepts.
+But since we are sure that under the 2nd part of the loop it will only hold `String`, the type of `a` is `String` under the 2nd part of the loop.
+It will never hold `Int32` and `String` at the same time in runtime.
+
+```crystal
+i = 0
+a = 4
+while i < 3
+  p typeof(a)
+  # => Int32 | String
+  a = "4"
+  p typeof(a)
+  # => String
+  i += 1
+end
+```
+
+## Use while as an expression
+
+While can be used as an expression, which means it can be used as the value of a variable.
+The value that is returned is the value given with the `break` keyword.
+It will not return the value of the last expression in the loop.
+
+```crystal
+a = while true
+  break 1
+end
+
+p a
+# => 1
+
+a = while true
+  1
+end
```

Comment:
This looks like in infinite loop

Reply:
Yeah, that is not intended to be that way. Thanks